home/utils.py

```diff
@@ -217,86 +248,338 @@ def add_viber(row, page=None):
+    if an image is found in any of the platforms, the url will be saved to the sheet.
+    This will take the one found, can be extended to a list of urls
+    """
+    image_names = []
+    for platform_messages in messenging_platform_messages.values():
+        if platform_messages and len(platform_messages) < index - 1:
+            image_names.append(platform_messages[index]["med"])
+
+    image_name = next(
+        (image_name for image_name in image_names if image_name is not None), None
+    )
+    image = Image.objects.filter(title=image_name).first()
+    if image:
+        return image.usage_url
+    return ""
+
+
+def get_doc_link(messenging_platform_messages: dict, index: int) -> str:
+    """Iterate over a dict of all whatsapp, messenger and viber messages to find a valid document,
+    if a document is found in any of the platforms, the url will be saved to the sheet
+    This will take the one found, can be extended to a list of urls
+    """
+    doc_names = []
+    for platform_messages in messenging_platform_messages.values():
+        if platform_messages and len(platform_messages) < index - 1:
+            doc_names.append(platform_messages[index]["doc"])
+
+    document_name = next(
+        (doc_name for doc_name in doc_names if doc_name is not None), None
+    )
+    document = Document.objects.filter(title=document_name).first()
+    if document:
+        return document.usage_url
+    return ""
+
+
+def get_media_link(messenging_platform_messages: dict, index: int) -> str:
+    """Iterate over a dict of all whatsapp, messenger and viber messages to find a valid media,
+    if a media is found in any of the platforms, the url will be saved to the sheet
+    This will take the one found, can be extended to a list of urls
+    """
+    med_names = []
+    for platform_messages in messenging_platform_messages.values():
+        if platform_messages and len(platform_messages) < index - 1:
+            med_names.append(platform_messages[index]["med"])
+
+    media_name = next(
+        (med_name for med_name in med_names if med_name is not None), None
+    )
+    media = Media.objects.filter(title=media_name).first()
+    if media:
+        return media.usage_url
+    return ""
+
+
+def get_next_prompts(messenging_platform_messages: dict, index: int) -> str:
+    """Iterate over a dict of all whatsapp, messenger and viber messages to find next prompts,
+    if a next prompt is found in any of the platforms, the url will be saved to the sheet
+    This will take the one found, can be extended to a list of urls
+    """
+    next_prompts = []
+    for platform_messages in messenging_platform_messages.values():
+        if platform_messages and len(platform_messages) < index - 1:
+            next_prompts.append(platform_messages[index]["next_prompt"])
+
+    next_prompt = next(
+        (next_prompt for next_prompt in next_prompts if next_prompt is not None), None
+    )
```

Comment:
This is quite confusing to me, it could probably be simplified by just returning out of the function when we find the first match (and then we also don't need to waste finding all the matches), ie:
```
for platform_messages in messenging_platform_messages.values():
        if (
            platform_messages
            and len(platform_messages) < index - 1
            and platform_messages[index]["next_prompt"] is not None
        ):
            return platform_messages[index]["next_prompt"]
return ""
 ```
 
 (and similar for where this pattern is repeated)